tests/test_recipient_csv.py

```diff
@@ -422,6 +422,21 @@ def test_overly_big_list_stops_processing_rows_beyond_max(mocker):
     ) == 10
 
 
+def test_rows_processing_timeout():
+    recipients = RecipientCSV(
+        # Keep the number of rows small to avoid DoSing other tests
+        'phone_number,07900900900\n' * 1000,
+        template=_sample_template('sms'),
+        # Use "0" to avoid flakey behaviour based on the no. of rows
+        processing_timeout=0,
```

Comment:
I did think of 0.1s, but this raises the ugly question of "how big do I need to go to ensure it takes >0.1s?" and the secondary question of "how much do I want to slow down the test suite because of that?". 

I think we are checking the two most important things here:

- The code runs normally most of the time (with the default timeout).
- The code errors if the timeout is low (0 being the extreme case).

Any other tests would be veering into checking the Thread class works as expected or benchmarking the code, which should be done elsewhere to avoid flakiness or slowing down the test suite.

Can you think of something I'm missing here?

Reply:
Good point about the other tests will make sure it works with the default timeout. That was something I'd missed.

Happy to stick with this! Thanks for explaining.